### PR TITLE
Notebook Objects cannot be created with CouchDB enabled

### PR DESCRIPTION
**Pull request by scottbell**
### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [X] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue?

### CODE REVIEW

**Comment by scottbell** [comment]
This PR addresses:
https://github.com/nasa/openmct/issues/4422
Note that though the null check seems to fix the immediate issue, I'm still a bit unclear of the telos for the monkey patch:
https://github.com/nasa/openmct/blob/master/src/plugins/notebook/monkeyPatchObjectAPIForNotebooks.js
Specifically, why is the monkeyPatch calling `destroyMutable` during its `save`?

**Comment by scottbell** [comment]
The Codecov Report is cool!!

File: src/plugins/persistence/couch/pluginSpec.js [hunk 1]
```diff
@@ -41,13 +41,12 @@ describe('the plugin', () => {
                 namespace: '',
                 key: 'some-value'
             },
-            type: 'mock-type',
+            type: 'notebook',
```
Comment (scottbell) [inline src/plugins/persistence/couch/pluginSpec.js:44]: changing type to `notebook` so observer gets called, which was the source of the bug.

File: src/plugins/persistence/couch/pluginSpec.js [hunk 2]
```diff
@@ -98,33 +97,26 @@ describe('the plugin', () => {
             fetch.and.returnValue(mockPromise);
         });
 
-        it('gets an object', () => {
-            return openmct.objects.get(mockDomainObject.identifier).then((result) => {
-                expect(result.identifier.key).toEqual(mockDomainObject.identifier.key);
-            });
+        it('gets an object', async () => {
```
Comment (scottbell) [inline src/plugins/persistence/couch/pluginSpec.js:100]: refactored tests to use async/await

**Review by akhenry — APPROVED** [review]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

**Pull request merged**